controllers/cluster_status.go

```diff
@@ -877,3 +881,35 @@ func getPodsTopology(
 
 	return apiv1.Topology{SuccessfullyExtracted: true, Instances: data}
 }
+
+// SetClusterConditionsInStatus is update condition in cluster status section,
+// it is based on number of pods in <cluster.spec.instance> is equal to cluster up/ready pods
+// then it will update condition type `Ready` with status `True` otherwise
+// with status `False`.
+func (r *ClusterReconciler) SetClusterConditionsInStatus(
+	ctx context.Context,
+	cluster *apiv1.Cluster,
+) {
+	contextLogger := log.FromContext(ctx)
+	podList, err := r.getManagedPods(ctx, cluster)
+	if err != nil {
+		contextLogger.Error(err, "pods are not found, skipping to set cluster conditions")
+	}
+	var conditions metav1.Condition
+	if cluster.Spec.Instances == utils.CountReadyPods(podList.Items) {
```

Comment:
could be already checking the status as it's already being set. https://github.com/cloudnative-pg/cloudnative-pg/pull/461/files#diff-5cd470cc9a2a125dc3413b4f557332272a815309dd2dbb25d182fcb7b80763ecR276

Reply:
@phisco Could you please provide more details.
I am not getting your point.